source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.cc

```diff
@@ -0,0 +1,56 @@
+#include "source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_metadata_file_reader.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+namespace {
+
+bool isIndirectionFile(const std::string& file_name) {
+  return absl::StartsWith(file_name, "dt_metadata_");
+}
+
+const std::string readFile(const std::string& file_name) {
```

Comment:
nit: top-level const doesn't do anything.
```suggestion
std::string readFile(const std::string& file_name) {
```

Reply:
Removed! [446a31d](https://github.com/envoyproxy/envoy/pull/30824/commits/446a31de3e2a01fd1ff64d74866b39ddef0bbd6b)